From 0c5f3af37ab6b38125984405ee022c0b1799e6a9 Mon Sep 17 00:00:00 2001
From: radhakrishna <radhakrishna@codeaurora.org>
Date: Fri, 31 Oct 2014 15:48:14 +0530
Subject: [PATCH] sf: Make sure HWC_BLENDING_NONE is set for opaque layer

Before configuring the layers to HWC, make sure
HWC_BLENDING_NONE is set if the layer is opaque

Change-Id: Ie84aa9d93e98a244692a0cffc2de653a9079f8b4
Crs-fixed: 957373
---
 services/surfaceflinger/Layer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index c346a2f..69e5c68 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -467,7 +467,11 @@ void Layer::setGeometry(
 
     // this gives us only the "orientation" component of the transform
     const State& s(getDrawingState());
+#ifdef QTI_BSP
+    if (!isOpaque(s)) {
+#else
     if (!isOpaque(s) || s.alpha != 0xFF) {
+#endif
         layer.setBlending(mPremultipliedAlpha ?
                 HWC_BLENDING_PREMULT :
                 HWC_BLENDING_COVERAGE);
